### PR TITLE
Client metrics & basic logging

### PR DESCRIPTION
**Pull request by leonbotros**
This PR adds basic logging using [`env_logger`](https://docs.rs/env_logger/latest/env_logger/) and metrics using [`prometheus`](https://crates.io/crates/prometheus). Some client information (host app, host app version, client, client version) is collected from a custom HTTP header (`X-Postguard-Client-Version`). The header value is expected to be a comma seperated list of the values mentioned above, anything else is ignored.

Before merging:
- [x] ~deal with resource exhaustion attacks (allocation)~,
- [x] are there more metrics we want to collect?
- [x] make sure no path parameters can end up in the metrics,

### CODE REVIEW

File: irmaseal-pkg/src/server.rs [hunk 1]
```diff
@@ -35,20 +52,33 @@ pub async fn exec(server_opts: ServerOpts) {
...
+
     HttpServer::new(move || {
         App::new()
+            .wrap(
+                Logger::new(
+                    "request=%{PATH}xi, status=%s, client=%{CLIENT_ID}xi, response_time=%D ms",
+                )
+                .custom_request_replace("CLIENT_ID", client_version)
+                .custom_request_replace("PATH", |req| {
+                    req.match_pattern().unwrap_or("-".to_string())
```
Comment (leonbotros) [inline irmaseal-pkg/src/server.rs:65]: I think it's better to not log paths that aren't matched. They might contain tokens.

File: irmaseal-pkg/src/middleware/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+use futures::Future;
+
+pub(crate) fn collect_metrics<
+    B: MessageBody,
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = actix_web::Error>,
+>(
+    req: ServiceRequest,
+    srv: &S,
+) -> impl Future<Output = Result<ServiceResponse<B>, actix_web::Error>> {
+    if let Some(Ok(header)) = req.headers().get(PG_CLIENT_HEADER).map(HeaderValue::to_str) {
```
Comment (leonbotros) [inline irmaseal-pkg/src/middleware/metrics.rs:17]: [If-let chains](https://rust-lang.github.io/rfcs/2497-if-let-chains.html) would be nice here.

File: irmaseal-pkg/src/middleware/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        let res = test::call_service(&app, req).await;
+        assert_eq!(res.status(), StatusCode::OK);
+        let body = test::read_body(res).await;
+
+        let expected ="\
+        # HELP postguard_clients Contains information about PostGuard clients connecting with the PKG.\n\
+        # TYPE postguard_clients counter\n\
+        postguard_clients{client=\"pg4ol\",client_version=\"0.0.1\",host=\"Outlook\",host_version=\"1234.5678.90\",path=\"/v2/key/{timestamp}\"} 1\n\
+        postguard_clients{client=\"pg4ol\",client_version=\"0.0.1\",host=\"Outlook\",host_version=\"1234.5678.90\",path=\"/v2/parameters\"} 2\n\
+        postguard_clients{client=\"pg4tb\",client_version=\"0.0.2\",host=\"Thunderbird\",host_version=\"1234.5678.90\",path=\"/v2/parameters\"} 1\n";
```
Comment (leonbotros) [inline irmaseal-pkg/src/middleware/metrics.rs:116]: I changed this to a multi-line string for readability.

**Review by leonbotros — COMMENTED** [review]
Formatting

File: irmaseal-pkg/src/server.rs (hunk 1 above)
Comment (leonbotros) [reply irmaseal-pkg/src/server.rs:65]: According to the documentation, "-" is convention.

**Review by Iso5786 — APPROVED** [review]

**Pull request merged**